NameLayerMain/src/vg/civcraft/mc/namelayer/database/GroupManagerDao.java

```diff
@@ -546,16 +546,29 @@ public void initializeStatements(){
 		
 		getAllGroupIds = db.prepareStatement("select group_id from faction_id");
 	}
+	
+	public synchronized void dbrefresh(){
+		NameLayerPlugin.reconnectAndReintializeStatements();
+	}
+	
+	public synchronized ResultSet queryStatement(PreparedStatement stat) throws SQLException{
+		return stat.executeQuery();
```

Comment:
Yes. These actions also need to be synchronized for three reasons. One because if the db is reconnected all prepared statements are re prepared. Two because prepared statements are not thread safe, this is used to make sure only one is used at a time and if a thread is waiting to use the same one it doesn't mutate. And three only a single prepared statement may be run on a connection at a time. Essentially it enables non synced calls to be queued until the main thread is finished with it.